ssl/ssl_lib.c

```diff
@@ -965,11 +965,21 @@ int SSL_set_trust(SSL *s, int trust)
 
 int SSL_set1_host(SSL *s, const char *hostname)
 {
+    /* If a hostname is provided and parses as an IP address,
+     * treat it as such. */
+    if (hostname && X509_VERIFY_PARAM_set1_ip_asc(s->param, hostname) == 1)
+        return 1;
+
     return X509_VERIFY_PARAM_set1_host(s->param, hostname, 0);
 }
 
 int SSL_add1_host(SSL *s, const char *hostname)
 {
+    /* If a hostname is provided and parses as an IP address,
+     * treat it as such. */
+    if (hostname && X509_VERIFY_PARAM_set1_ip_asc(s->param, hostname) == 1)
```

Comment:
This looks like a mistake to me. A second IP address via `add1` should not clobber the first. Here, if an IP address is passed to `add1_host` an error should be returned instead.

Reply:
It ends up being slightly non-trivial and clumsy but I think commit 5d6a26cfff  should achieve this.